### PR TITLE
Migrate `Team.GetMembers` to Sync by default

### LINKED ISSUE #11122: Migrate "Team.GetMembers" to Sync by default
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-16177)

Mattermost is migrating its Store layer to be sync by default, and only use Async when needed and we're looking for contributors to help with that effort. This Help Wanted issue is to migrate the `GetMembers` in the `Team` store.

The expected way to implement it is, go to the `Team` store implementation in the `store/sqlstore/` directory, modify the method `GetMembers` to return directly an object from the `model` module, and a `*model.AppError` (removing the `store.Do` wrapper). After that, you must modify the interface defined in `store/store.go` to match with the changes. Then, you should execute `make store-mocks` to rebuild the mocks with the new interface. And finally, modify the rest of the code (tests included) to use the new interface of the function properly.

Example: mattermost/mattermost-server#10613

### PR DESCRIPTION
**Pull request by rvillablanca**
Fixes #11122. GetMembers is sync now

### CODE REVIEW

**Comment by rvillablanca** [comment]
@jespino I have run with success `make build-linux`, `make check-style`, and the changes are no very invasive, so I don't understand why this PR is failing. Could you help my with this please? Also PR #10971 has problems

**Comment by rvillablanca** [comment]
I still do not understand why this sometimes fails :(

**Comment by hanzei** [comment]
Me neither. But now it's green  :wink:

**Comment by rvillablanca** [comment]
What about #10971, could you check why that PR is failing @hanzei ?

**Comment by jespino** [comment]
@rvillablanca yes, we have some problems with the CI I think, there are flaky "behavior" on the tests, they fails time to time and we need to figure out why.

File: app/team.go [hunk 1]
```diff
@@ -709,11 +709,11 @@ func (a *App) GetTeamMembersForUserWithPagination(userId string, page, perPage i
 }
 
 func (a *App) GetTeamMembers(teamId string, offset int, limit int, restrictions *model.ViewUsersRestrictions) ([]*model.TeamMember, *model.AppError) {
-	result := <-a.Srv.Store.Team().GetMembers(teamId, offset, limit, restrictions)
-	if result.Err != nil {
-		return nil, result.Err
+	members, err := a.Srv.Store.Team().GetMembers(teamId, offset, limit, restrictions)
```
Comment (jespino) [inline app/team.go]: Here you can just `return a.Srv.Store.Team().GetMembers(teamId, offset, limit, restrictions)`

**Pull request merged**